### DISCUSSION TITLE
margin notes not appearing where desired

### DISCUSSION THREAD

**Discussion by ericpgreen** [quarto-dev/quarto-cli#10109] [Q&A]
### Description

Is it possible to have a margin note appear in the margin at the start of a paragraph, taking priority over margin references cited in the same paragraph?

In the following example, I'd like the note to appear in the position of the reference, followed by the reference.

<img width="1002" alt="Screenshot 2024-06-19 at 10 35 18 AM" src="https://github.com/quarto-dev/quarto-cli/assets/546433/809a2acb-166f-4e20-9c06-f56b50ebc65d">

To reproduce, start a new Quarto book project in RStudio and add two elements:

**(1) add the following text to `index.qmd` file** below the pre-existing code block.

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque blandit porta massa, eget vehicula est imperdiet ut. Mauris elementum nisi consequat fermentum luctus. Mauris vel arcu vitae nibh molestie porttitor. Integer vel pellentesque nulla. Suspendisse potenti. Donec ipsum turpis, facilisis sed neque ut, congue bibendum erat. Aliquam erat volutpat. Sed hendrerit turpis nec porttitor bibendum. Aenean vel porttitor neque.

Nam sit amet eros arcu. Mauris vitae sem nisi. Duis interdum mauris vitae varius congue. Curabitur at sem dignissim, porta lacus quis, rutrum urna. Donec tellus lacus, aliquam vitae elit at, pellentesque commodo velit. Sed malesuada velit eget orci rutrum, non vestibulum arcu fermentum. Sed id elit pretium, ultrices nunc eget, ornare lorem. Phasellus tincidunt fermentum justo tempus tincidunt. Suspendisse consequat vehicula lacinia. Etiam pharetra nisl in dui aliquam, vitae porttitor neque pellentesque. Donec et scelerisque ante.[@knuth84]

:::{.column-margin}
I want to put a note level with the start of the paragraph that begins with "Nam sit amet eros arcu".
:::
```

**(2) YAML**

Add `citation-location: margin` to the YAML in `_quarto.yml`.

ps. In the past, I believe placing `.column-margin` ABOVE the paragraph led to the desired behavior. 

```
R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.1
```

```
quarto::quarto_version()
#[1] ‘1.4.555’
```

Posted originally at https://forum.posit.co/t/margin-notes-not-appearing-where-desired/188317?u=eric_green

**Comment by mcanouil** [comment]
To be sure I understand. You are placing a custom margin after a reference but want the reference to be after that custom margin text?
To me, everything works as expected and as you "asked" (with the code you wrote).

I don't believe this was ever possible or ever will.

Note that footnotes would behave more like references.

Additionally, instead of using div for margin, you could try with span to make your note tied to the same paragraph. https://quarto.org/docs/authoring/markdown-basics.html#sec-divs-and-spans

**Reply by ericpgreen** [reply]
Thanks @mcanouil. I'm looking to have a margin note appear in the margin aligned with the first line of the paragraph. Citations in that paragraph should also appear in the margin, but below the note. I think this used to be the default behavior (see below) but now when I render the note appears below the references.

In this [old example](https://ericpgreen.github.io/ghr_ebook/ghr.html#clinical-research), the note appears before the reference:

<img width="1290" alt="Screenshot 2024-06-24 at 10 08 00 AM" src="https://github.com/quarto-dev/quarto-cli/assets/546433/3a91413b-6840-48ab-ad47-b7065f1d99ac">

**Reply by mcanouil** [reply]
Are you using the same code?
Could you check what Quarto version was used? (It should at least be in the HTML metadata)

**Reply by ericpgreen** [reply]
Yeah, same code. I last rendered with `quarto-1.2.335`. In the second example I shared, the reference does not render today, just the note.

Here's what I'm trying to achieve. I think what you're saying is that this should not be possible. Is that right?

<img width="1002" alt="new" src="https://github.com/quarto-dev/quarto-cli/assets/546433/cc029fa8-e712-46e5-a7f5-4cfe9c0c9bf0">

**Reply by mcanouil** [reply]
If the behaviour was changed, it's likely because of a more important issue.
We'll need to investigate more.
What's certain right now is that the order is defined by the order in which references/notes appear.

**Reply by ericpgreen** [reply]
Thanks for your feedback. 

I'm investigating spans as you suggested. From the [docs](https://quarto.org/docs/authoring/article-layout.html#asides) it looks like `[some note]{.aside}` should work, but it's rendering in the main column as normal text for me. 

<img width="1013" alt="Screenshot 2024-06-24 at 11 11 56 AM" src="https://github.com/quarto-dev/quarto-cli/assets/546433/8a077584-e4e0-47e6-85eb-7f0dcc02fec9">

Should this get me closer to what I'm looking for? 

```
[I want to put a note level with the start of the paragraph that begins with "Nam sit amet eros arcu".]{.aside}

Nam sit amet eros arcu. Mauris vitae sem nisi. Duis interdum mauris vitae varius congue. Curabitur at sem dignissim, porta lacus quis, rutrum urna. Donec tellus lacus, aliquam vitae elit at, pellentesque commodo velit. Sed malesuada velit eget orci rutrum, non vestibulum arcu fermentum. Sed id elit pretium, ultrices nunc eget, ornare lorem. Phasellus tincidunt fermentum justo tempus tincidunt. Suspendisse consequat vehicula lacinia. Etiam pharetra nisl in dui aliquam, vitae porttitor neque pellentesque. Donec et scelerisque ante.[@knuth84]
```

**Reply by ericpgreen** [reply]
I think asides are broken in my version of Quarto per [this issue](https://github.com/quarto-dev/quarto-cli/issues/10018). I will investigate how to update Quarto and try again (I have the version bundled with RStudio).
